engine/src/main/java/org/kakara/engine/item/MeshGameItem.java

```diff
@@ -245,5 +249,39 @@ public GameItem clone(boolean exact) {
         return clone;
     }
 
+    public void movePosition(float offsetX, float offsetY, float offsetZ) {
```

Comment:
What in the world? Why are you working with sin and cos? Shouldn't you just add the offset to the positon?